src/pages/Home/index.tsx

```diff
@@ -1,14 +1,40 @@
 import { Container } from 'react-bootstrap';
-
+import { useEffect, useState } from 'react';
+import { OrderProps, BobaProps } from 'types/common/main';
+import { bobaList } from 'assets/sampleBobaAPI';
 import BobaContainer from 'components/BobaContainer';
-import { OrderProps } from 'types/common/main';
+import SearchBar from 'components/SearchBar';
 
 const Homepage = (props: OrderProps): JSX.Element => {
   const { order, setOrder } = props;
+
+  const [bobaListx, setBobaListx] = useState<BobaProps[]>(bobaList);
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  const filteredfunction = (bobaList: BobaProps[], query: string): BobaProps[] => {
+    return bobaList.filter((filtered: BobaProps) =>
+      filtered.name.toLowerCase().includes(query.toLowerCase()),
+    );
+  };
+
+  useEffect(() => {
+    if (searchQuery) {
+      const filterlist = filteredfunction(bobaList, searchQuery);
+      if (!filterlist) {
+        setBobaListx([]);
+      } else {
+        setBobaListx(filterlist);
+      }
+    } else {
+      setBobaListx(bobaList);
+    }
+  }, [searchQuery]);
```

Comment:
```
  useEffect(() => {
    if (filteredBobaList.length === 0) {
      setFilteredBobaList([]);
    } else {
      setFilteredBobaList(filteredBobaList);
    }
  }, [filteredBobaList]);
  ```

Reply:
also here, why filteredBobalist here, i originally put searchQuery cuz that is something i want to keep tracking of changing.

Reply:
If it's functional now, keep as it is. We can refactor for performance later.